src/Interpreters/Aggregator.cpp

```diff
@@ -1180,10 +1181,15 @@ Block Aggregator::prepareBlockAndFill(
             if (aggregate_functions[i]->isState())
             {
                 /// The ColumnAggregateFunction column captures the shared ownership of the arena with aggregate function states.
-                ColumnAggregateFunction & column_aggregate_func = assert_cast<ColumnAggregateFunction &>(*final_aggregate_columns[i]);
+                ColumnAggregateFunction * column_aggregate_func = nullptr;
+                /// Aggregate state can be wrapped into array if aggregate function ends with -Resample combinator.
+                if (auto * column_array = typeid_cast<ColumnArray *>(final_aggregate_columns[i].get()))
```

Comment:
Is it possible with other columns? Why don't use `IColumn::forEachSubcolumn`?

Reply:
Now it's impossible, but I agree, that `forEachSubcolumn` is better.

Reply:
Did you say impossible?
#12376